pyccel/ast/builtins.py

```diff
@@ -482,22 +482,17 @@ def __init__(self, *args):
         stop = None
```

Comment:
Since we are at it, I think this line is never used, because `stop` is always redefined.

I think it would be cleaner if we moved the other two variable definitions (for `start` and `step`) inside the `if/elif/else` block.

Reply:
I'm not sure I agree. These are default values. It is better to reduce code duplication where possible. I can remove `stop` if you want but in the context of default values it makes sense to provide all 3

Reply:
Since this code is not compiled but interpreted, my suggestion is faster because it creates the minimum number of Python objects in memory. But it is true that one (!) additional line was duplicated:
```python
n = len(args):

if n == 1:
    start = LiteralConstant(0)
    stop  = args[0]
    step  = LiteralConstant(1)
elif n == 2:
    start = args[0]
    stop  = args[1]
    step  = LiteralConstant(1)
elif n == 3:
    start = args[0]
    stop  = args[1]
    step  = args[2]
else:
    raise ...
```

Reply:
Ok. I have changed the code to match your suggestion

Reply:
Thanks!